packages/transactions/src/transaction.ts

```diff
@@ -164,10 +170,11 @@ export class StacksTransaction {
     if (isSingleSig(condition)) {
       condition.signature = nextSig;
     } else {
-      const compressed = bytesToHex(privateKey.data).endsWith('01');
       condition.fields.push(
         createTransactionAuthField(
-          compressed ? PubKeyEncoding.Compressed : PubKeyEncoding.Uncompressed,
+          privateKey.data.byteLength === PRIVATE_KEY_COMPRESSED_LENGTH
+            ? PubKeyEncoding.Compressed
+            : PubKeyEncoding.Uncompressed,
```

Comment:
Would be safer to have a `switch` here, and fail if `privateKey.data.byteLength` isn't `PRIVATE_KEY_COMPRESSED_LENGTH` or `PRIVATE_KEY_UNCOMPRESSED_LENGTH`